wallet/src/main/java/com/android/identity_credential/wallet/presentation/PresentmentFlow.kt

```diff
@@ -0,0 +1,203 @@
+import com.android.identity.securearea.software.SoftwareKeyInfo
+import com.android.identity.securearea.software.SoftwareKeyUnlockData
+import com.android.identity.securearea.software.SoftwareSecureArea
+import com.android.identity.trustmanagement.TrustPoint
+import com.android.identity.util.Constants
+import com.android.identity_credential.wallet.R
+import com.android.identity_credential.wallet.WalletApplication
+import com.android.identity_credential.wallet.ui.prompt.biometric.showBiometricPrompt
+import com.android.identity_credential.wallet.ui.prompt.consent.showConsentPrompt
+import com.android.identity_credential.wallet.ui.prompt.passphrase.showPassphrasePrompt
+
+const val TAG = "PresentmentFlow"
+
+/**
+ * Function responsible for showing the Presentment Flow for a given [DocumentRequest] and returning
+ * response bytes or throw an Exception. The purpose of the Presentment Flow is to authorize a
+ * [Document] that is suitable for fulfilling the specified [DocumentRequest], by showing a series
+ * of consent and authentication Prompts to the user and upon success the response bytes containing
+ * the [Document] are returned.
+ *
+ * The Presentment Flow always starts by showing the Consent Prompt to confirm with the user
+ * before sending sensitive data of a [Document] to a Verifying party. If the Consent Prompt
+ * is successful then the Biometric Prompt and/or Passphrase Prompt will be shown so long as the
+ * auth key is locked for their corresponding Secure Area.
+ *
+ * An [Exception] (mostly [IllegalStateException]) is thrown for numerous reasons, such as
+ * if credentials could not be found in a Document; authentication type was not specified when
+ * showing Biometric prompt; a (new) secure area has not been handled/implemented yet; if a prompt
+ * is unsuccessful because the user cancelled or was not able to authenticate, etc..
+ *
+ * @param activity the [FragmentActivity] used for showing Dialog Fragments.
+ * @param walletApp the [WalletApplication] instance used for dependencies.
+ * @param documentRequest a [DocumentRequest] instance defining a list of
+ *      credential fields/data elements to obtain from the suitable [Document] in MdocCredential.
+ * @param mdocCredential the object containing the [Document] and docType amongst other properties.
+ * @param trustPoint if provided, identifies the Verifying party.
+ * @param encodedSessionTranscript the bytes of `SessionTranscript` CBOR.
+ * @return the response bytes that contain the suitable [Document], else throws an exception.
+ * @throws Exception for incorrect configurations, cannot find credentials or unsuccessful
+ *      prompts because user cancelled or wasn't able to authenticate.
+ */
+suspend fun showPresentmentFlow(
+    activity: FragmentActivity,
+    walletApp: WalletApplication,
+    documentRequest: DocumentRequest,
+    mdocCredential: MdocCredential,
+    trustPoint: TrustPoint?,
+    encodedSessionTranscript: ByteArray
+): ByteArray {
```

Comment:
As discussed this needs to return the bytes of the Document CBOR according to 18013-5. That's actually already what the kdoc says but the code does something else... should be a very simple change!

Reply:
I believe this code is already doing that!

Once user succeeds the Consent Prompt on line 79, it uses the`deviceResponseGenerator` on line 85
```
val deviceResponseGenerator = DeviceResponseGenerator(Constants.DEVICE_RESPONSE_STATUS_OK)
```
to add the Document on line 108 after successfully passing all Prompts when trying to sign the Document data with Credentials and KeyUnlockData 

If successful because the auth key is unlocked for the secure area the credential is in (and it's not throwing a KeyLockedException). then finally returns `the bytes of DeviceResponse CBOR` on line 109

```
return deviceResponseGenerator.generate()
```

Reply:
I'm talking about the *Document* CBOR and you are talking about the *DeviceResponse* CBOR. They are different things. Each *DeviceResponse* has a number of *Document* instances, one for each of the mdocs that the user sends.